exasol/analytics/schema/column_name_builder.py

```diff
@@ -37,6 +39,8 @@ def with_table_like_name(
         return self
 
     def build(self) -> ColumnName:
+        if self._name is None:
+            raise TypeCheckError("Name must not be None.")
```

Comment:
Is it OK to raise a `TypeCheckError` here (same as typeguard does)?
Raising a different Error such as `ValueError` would require to update the unit tests.

Reply:
it is not  a typeguard typecheck, so I would create my own TypeCheckError. However, we can do this also in another PR.